builder/git/membership/gitea/member.go

```diff
@@ -0,0 +1,151 @@
+package gitea
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/OpenCSGs/gitea-go-sdk/gitea"
+	"opencsg.com/csghub-server/builder/git/membership"
+	"opencsg.com/csghub-server/common/utils/common"
+)
+
+var _ membership.GitMemerShip = (*Client)(nil)
+
+func (c *Client) AddRoles(ctx context.Context, org string, roles []membership.Role) error {
+	var errs error
+	for _, to := range c.getTargetOrgs(org) {
+		errs = errors.Join(errs,
+			c.addRoles(ctx, to, roles))
+	}
+	return errs
+}
+
+func (c *Client) addRoles(ctx context.Context, org string, roles []membership.Role) error {
+	var errs error
+	for _, role := range roles {
+		opt := c.getTeamOptByRole(role)
+		_, _, err := c.giteaClient.CreateTeam(org, opt)
```

Comment:
Will this throw an error if the team already exists in gitea.

Reply:
maybe, but to new org, there should be no team exists